### ISSUE TITLE
failure to follow link with accents

### ISSUE THREAD

**Issue by anarcat** [linkchecker/linkchecker#644]
I encountered a bug similar to #451 on my own blog. if you run this command:

```
linkchecker --check-extern --no-robots --recursion-level 1 https://anarc.at/blog/2021-11-21-mbsync-vs-offlineimap
```

... on top of a bunch of other crap, you'll find:

```
URL                   `../2021-12-22-internet-nÃ©ocolonial/'
URL parente           https://anarc.at/blog/2021-11-21-mbsync-vs-offlineimap/, ligne 55, col. 11
URL Réelle            https://anarc.at/blog/2021-12-22-internet-n%C3%83%C2%A9ocolonial/
Temps de vérification 3.263 secondes
Taille                8KB
Résultat              Erreur: 404 Not Found
10 threads active,    55 links queued,   20 links in  86 URLs checked, temps d'exécution 6 secondes

```

well that's in french, but the point is it's got mojibake here, the proper URL is `../2021-12-22-internet-néocolonial/`, and it works in a web browser (e.g. firefox). i must admit it's kind of bold of me to use accents in URLs like this, but fuck it, it's 2020(+) that shit should work now. :)

This is with linkchecker 10.0.1-2+b1 on Debian bookworm/testing (which reminds me someone (probably me!) needs to update linkchecker in debian, it seems like).

_Originally posted by @anarcat in https://github.com/linkchecker/linkchecker/issues/451#issuecomment-1235717137_

**Comment by anarcat** [comment]
(the other garbage is also interesting... linkchecker is stumbling upon this link:

URL                   `https://pyropus.ca./software/getmail/'
Nom                   `getmail'
URL parente           https://anarc.at/blog/2021-11-21-mbsync-vs-offlineimap/, ligne 1249, col. 5
URL Réelle            https://pyropus.ca./software/getmail/
Temps de vérification 2.539 secondes
Taille                322B
Info                  Redirected to
                      `https://pyropus.ca./software/getmail/'.
Résultat              Erreur: TooManyRedirects: Exceeded 10 redirects.

... which eventually settles in firefox, to https://pyropus.ca./software/getmail/, without redirects, from what i understand. and yes, that's a valid URL, even if a little annoying...

the other one is the w3.org validator which, somehow, we fail to follow correctly. maybe i should have a nofollow or something to keep the bot out of there, not sure.

end of parenthesis)

**Comment by anarcat** [comment]
here's the rather verbose output of `-Dchecking`: [debug.txt](https://github.com/linkchecker/linkchecker/files/9480407/debug.txt)

my locale:

```
anarcat@emma:~[1]$ locale
LANG=fr_CA.UTF-8
LANGUAGE=
LC_CTYPE="fr_CA.UTF-8"
LC_NUMERIC="fr_CA.UTF-8"
LC_TIME="fr_CA.UTF-8"
LC_COLLATE="fr_CA.UTF-8"
LC_MONETARY="fr_CA.UTF-8"
LC_MESSAGES="fr_CA.UTF-8"
LC_PAPER="fr_CA.UTF-8"
LC_NAME="fr_CA.UTF-8"
LC_ADDRESS="fr_CA.UTF-8"
LC_TELEPHONE="fr_CA.UTF-8"
LC_MEASUREMENT="fr_CA.UTF-8"
LC_IDENTIFICATION="fr_CA.UTF-8"
LC_ALL=
```

**Comment by anarcat** [comment]
> First step is to upgrade to v10.1.0, it includes http character encoding fixes [c89c617](https://github.com/linkchecker/linkchecker/commit/c89c617a58b610f3aadfefd9b2d7efdc26681f52) and [a042144](https://github.com/linkchecker/linkchecker/commit/a04214465a9952334ddce9eec0e1732ff40ecfac).

okay, so that fixed it. i'm going to update the debian package so everyone gets those goodies.

i think it's safe to ignore the other stuff, if requests fail, not sure we want to go down those weeds...

**Comment by anarcat** [comment]
> i think it's safe to ignore the other stuff, if requests fail, not sure we want to go down those weeds...

re the `pyropus.ca.` bug, i foud this was a good tutorial of the issue:

https://jvns.ca/blog/2022/09/12/why-do-domain-names-end-with-a-dot-/

... case someone wants to followup with upstream (requests).
